### ISSUE TITLE
[BUG]: App is being attributed to wrong environment (Wrong number of apps shown)

### ISSUE THREAD

**Issue by SteveAsch** [microsoft/powerapps-tools#626]
In the COE starter kit environment when I open the environment (Monitor > Environments) the app count is 1
If I open the environment in the Power Platform Admin center (admin.powerplatform.microsoft.com)> resources > power apps it shows 4.

COE 1.7.1.7
![image](https://user-images.githubusercontent.com/47640245/101800607-517ac980-3ac2-11eb-9288-8489bb79385d.png)
COE Screenshot

PP Admin Screenshot
![image](https://user-images.githubusercontent.com/47640245/101800744-753e0f80-3ac2-11eb-9a07-b0dbeac8891b.png)

thoughts?
Steve

**Comment by JeneferM-MSFT** [comment]
Hello. Apps in dev sku environments cannot be tallied in the toolkit by default. 
Please see our [limitations](https://docs.microsoft.com/power-platform/guidance/coe/limitations#developer-environments-from-the-power-apps-community-plan)

Thanks for using CoE

**Comment by SteveAsch** [comment]
This is one environment example. It happens in most of the others as well. Please reopen this issue

**Comment by SteveAsch** [comment]
Here's the view in COE for the Prod environment. 5 Apps
![image](https://user-images.githubusercontent.com/47640245/101803508-758bda00-3ac5-11eb-9848-552bfe7bc53e.png)
And the same environment in power platform
![image](https://user-images.githubusercontent.com/47640245/101803615-93593f00-3ac5-11eb-9f69-cd71cdae0ada.png)

**Comment by SteveAsch** [comment]
Please reopen this case, not done yet

Steve

**Comment by SteveAsch** [comment]
I cant add to this case anymore online

**Comment by SteveAsch** [comment]
Awesome, thanks

**Comment by JeneferM-MSFT** [comment]
Can you please open the apps table in Excel (export or open in Excel)
![image](https://user-images.githubusercontent.com/62659043/101807915-85f28380-3aca-11eb-80fa-5f62964d2f55.png)

From here please
1) Filter to the envioronment where you expect to see the app, and see what the count is in the table
2) Remove the environment filter from 1, and now filter to the app, see if its there but attributed to the incorrect environment.

**Comment by SteveAsch** [comment]
There are 9 listed

App Display Name
--
Sales Hub
Customer   Service Hub
Solution Health Hub
Client   License Mgmt
ITK Service Request App
Sales   Team Member
Portal Management
ITK   DEV STEVE TEST
Customer Service workspace

**Comment by SteveAsch** [comment]
There are 5 listed

App Display Name
--
Solution Health Hub
Dynamics   365 App for Outlook
Customer Service workspace
Sales   Team Member
BizView

**Comment by JeneferM-MSFT** [comment]
Sorry I am going to need you to be more verbose. Let's stick with this one test environment
![image](https://user-images.githubusercontent.com/62659043/101813086-1df36b80-3ad1-11eb-9974-0865c219c9c1.png)


It sounds like you are saying that there are 7 apps in the admin.powerapps.com view of this envt
And when you opened the apps table, you only saw 5 for this envt. 

Is that accurate? 

If so, can you please look for the other 2 apps in the table? We need to see if they are there but mis-attributed to another environment. 

There is currently a product bug which has resulted in this for one of our customers. So I'm wanting to see if you ahve the same bug.

**Comment by SteveAsch** [comment]
Yes, that is accurate. 
**Sales Hub** and **Customer Service Hub** are the missing apps from this production environment
They are both listed in the development environment where they also exist as well as they should.
We move solutions from Dev to Prod so they should be listed there.

**Comment by JeneferM-MSFT** [comment]
Two more asks:
1) back to the 438... envt, can you please get the appid for those two missing apps? 
in the admin resource view, if you click the ..., share, the appid is embedded in the share url, in the black space shown here
![image](https://user-images.githubusercontent.com/62659043/101814689-56944480-3ad3-11eb-8b5a-eeb36d902376.png)

look for these guids in the app table. 

2) in the view of 438 itself, the list of apps, can you hit the ... drop down there for the two apps not in the CoE? is play enabled?

**Comment by SteveAsch** [comment]
For **customer service hub** I was able to find a match. It's listed as connected with the dev environment in the excel file
The **Sales Hub** is the same. The App ID in the excel file is also mapped to the dev environment.

**Comment by JeneferM-MSFT** [comment]
ok. you have found the same product bug then as well. 

I will send this link to the PM internally who is investigating.

**Comment by SteveAsch** [comment]
Thank you

**Comment by SteveAsch** [comment]
@JeneferM-MSFT Do we have an update for this yet? My client is also getting wrong reporting of apps.

**Comment by JeneferM-MSFT** [comment]
We are doing a re-architecture of the toolkit to the modern CDS Connector. That is likely will be how this is resolved for you all.  Excpected sometime in January.

**Comment by JeneferM-MSFT** [comment]
We have just released a version today with a major rewrite for reliability of the inventory suite. 
Be sure to follow the new steps for [installing updates](https://docs.microsoft.com/power-platform/guidance/coe/setup#installing-updates)
And thanks for using CoE!

**Comment by SteveAsch** [comment]
Installed update, ran Admin | Sync Template v3 flow. Same issue, was not fixed

**Comment by JeneferM-MSFT** [comment]
Per your other thread it sounds like update did not work. 
Are you saying that it worked?

**Comment by SteveAsch** [comment]
Hi,

Sorry for the confusion, here’s what happened.

I installed the update and it presented an error. I thought it failed because of something I must have done so I closed the error and clicked away. I did not note the error.

 

I went back to try again and it showed that it had in fact applied the update. I turned on the flows and ran the main flow. The core components and the audit logs were the only 2 solutions I installed updates for.

 

I’m working on other projects so I troubleshoot CoE when I can.  Due to the high volume of issues with CoE and instability of the application it’s had a larger than expected take of my available time.

 

My gut feel is that there’s a timeout or timing issue with flows that hasn’t been fully tested. Often I can resubmit a failed flow and it almost always works. Although support via github is fast it takes a lot of time to go through message board support. I would love a call for somebody to look at my configuration.

 

The app looks cool and I think it’s something I want to use with every client so I’m still trying! 

Steve
